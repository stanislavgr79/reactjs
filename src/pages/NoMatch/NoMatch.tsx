import React from 'react';

import Header from '../../components/structure/Header';
import Footer from '../../components/structure/Footer';

import './NoMatch.scoped.scss';
import { useLocation } from 'react-router-dom';
import { useSelector } from 'react-redux';
import { AppState } from '../../redux/redux-store';

export default function NoMatch(): JSX.Element {
  const unknownPath = useLocation();
  const { location } = useSelector((store: AppState) => {
    return store.searchStore;
  });
  return (
    <div className="nomatch-page">
      <Header />
      <div className="nomatch-page-panel">
        <div id="nomatch-header">
          <div id="title-heading" className="pagetitle">
            <h1 id="title-text">Page Not Found</h1>
          </div>
        </div>
        <div id="sidebar-container"></div>
        <p>We can&apos;t find that page. This could be because:</p>
        <ul>
          <li>The page {location != '' ? location : unknownPath.pathname} doesn&apos;t exist.</li>
          <li>The page exists, but you don&apos;t have view permission for that space.</li>
        </ul>
        <div id="action-messages">
          <h3>Go to</h3>
          <ul>
            <li>
              <a href="/movies">Site Homepage</a>
            </li>
          </ul>
        </div>
      </div>
      <Footer />
    </div>
  );
}
