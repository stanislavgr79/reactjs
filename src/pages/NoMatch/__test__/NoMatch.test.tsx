import React from 'react';
import { render } from '@testing-library/react';

import NoMatch from '../NoMatch';
import ReactRouter from 'react-router';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';

jest.mock('../../../components/structure/Header', () => {
  return {
    __esModule: true,
    default: () => {
      return <div className="header"></div>;
    },
  };
});
jest.mock('../../../components/structure/Footer', () => {
  return {
    __esModule: true,
    default: () => {
      return <div className="footer"></div>;
    },
  };
});

describe('NoMatch', () => {
  test('snapshot renders location is empty', () => {
    const initialState = { searchStore: { location: '' } };
    const mockStore = configureStore();
    const store = mockStore(initialState);

    const mockLocation = {
      pathname: '/movies',
      search: '',
      state: null,
      hash: '',
    };

    jest.spyOn(ReactRouter, 'useLocation').mockReturnValue(mockLocation);
    const tree = render(
      <Provider store={store}>
        <NoMatch />
      </Provider>,
    );

    expect(tree.baseElement).toMatchSnapshot();
    expect(tree.baseElement).toContainHTML('<div class="nomatch-page">');
    expect(tree.getByText(`The page /movies doesn't exist.`)).toBeInTheDocument();
  });

  test('snapshot renders location is 123456', () => {
    const initialState = { searchStore: { location: '123456' } };
    const mockStore = configureStore();
    const store = mockStore(initialState);

    const mockLocation = {
      pathname: '/movies',
      search: '',
      state: null,
      hash: '',
    };

    jest.spyOn(ReactRouter, 'useLocation').mockReturnValue(mockLocation);
    const tree = render(
      <Provider store={store}>
        <NoMatch />
      </Provider>,
    );

    expect(tree.baseElement).toMatchSnapshot();
    expect(tree.baseElement).toContainHTML('<div class="nomatch-page">');
    expect(tree.getByText(`The page 123456 doesn't exist.`)).toBeInTheDocument();
  });
});
