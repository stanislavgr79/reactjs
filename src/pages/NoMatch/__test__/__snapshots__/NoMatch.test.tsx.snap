// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NoMatch snapshot renders location is 123456 1`] = `
<body>
  <div>
    <div
      class="nomatch-page"
    >
      <div
        class="header"
      />
      <div
        class="nomatch-page-panel"
      >
        <div
          id="nomatch-header"
        >
          <div
            class="pagetitle"
            id="title-heading"
          >
            <h1
              id="title-text"
            >
              Page Not Found
            </h1>
          </div>
        </div>
        <div
          id="sidebar-container"
        />
        <p>
          We can't find that page. This could be because:
        </p>
        <ul>
          <li>
            The page 
            123456
             doesn't exist.
          </li>
          <li>
            The page exists, but you don't have view permission for that space.
          </li>
        </ul>
        <div
          id="action-messages"
        >
          <h3>
            Go to
          </h3>
          <ul>
            <li>
              <a
                href="/movies"
              >
                Site Homepage
              </a>
            </li>
          </ul>
        </div>
      </div>
      <div
        class="footer"
      />
    </div>
  </div>
</body>
`;

exports[`NoMatch snapshot renders location is empty 1`] = `
<body>
  <div>
    <div
      class="nomatch-page"
    >
      <div
        class="header"
      />
      <div
        class="nomatch-page-panel"
      >
        <div
          id="nomatch-header"
        >
          <div
            class="pagetitle"
            id="title-heading"
          >
            <h1
              id="title-text"
            >
              Page Not Found
            </h1>
          </div>
        </div>
        <div
          id="sidebar-container"
        />
        <p>
          We can't find that page. This could be because:
        </p>
        <ul>
          <li>
            The page 
            /movies
             doesn't exist.
          </li>
          <li>
            The page exists, but you don't have view permission for that space.
          </li>
        </ul>
        <div
          id="action-messages"
        >
          <h3>
            Go to
          </h3>
          <ul>
            <li>
              <a
                href="/movies"
              >
                Site Homepage
              </a>
            </li>
          </ul>
        </div>
      </div>
      <div
        class="footer"
      />
    </div>
  </div>
</body>
`;
